### PR TITLE
fix: better typescript code, use svg components

### LINKED ISSUE #18017: Empty snapshots diff between Linux/MacOS and Windows
- https://github.com/ant-design/ant-design/pull/17903#issuecomment-516774413
- https://github.com/ant-design/ant-design/pull/18006/files#diff-efa591f057303d0aee87cd2a1e6a6f88L1234
- https://github.com/ant-design/ant-design/pull/17858/files#diff-efa591f057303d0aee87cd2a1e6a6f88L1221

### PR DESCRIPTION
**Pull request by chenshuai2144**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #18017

### 💡 Background and solution

修复一下 ts 的类型定义，更加优雅。
顺便解决一下 svg 在 window 换行符造成 测试失败的问题

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      无需   |
| 🇨🇳 Chinese |      无需     |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
ci broken

**Comment by afc163** [comment]
image

**Comment by afc163** [comment]
image

**Comment by chenshuai2144** [comment]
改了好多，有点慌

File: components/empty/empty.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+export default () => (
+  <svg width="184" height="152" viewBox="0 0 184 152" xmlns="http://www.w3.org/2000/svg">
```
Comment (zombieJ) [inline components/empty/empty.tsx]: 是否应该弄个 `shouldComponentUpdate` 优化一下性能？

Comment (ztplz) [reply components/empty/empty.tsx]: 9102了  memo

**Pull request merged**